bootstrap/bootstrap.go

```diff
@@ -1216,6 +1216,10 @@ func (b *Bootstrap) updateGitMirror() (string, error) {
 		b.shell.Commentf("Cloning a mirror of the repository to %q", mirrorDir)
 		flags := "--mirror " + b.GitCloneMirrorFlags
 		if err := gitClone(b.shell, flags, b.Repository, mirrorDir); err != nil {
+			b.shell.Commentf("Removing mirror dir %q due to failed clone", mirrorDir)
+			if err := os.RemoveAll(mirrorDir); err != nil {
+				b.shell.Errorf("Failed to remove \"%s\" (%s)", mirrorDir, err)
+			}
 			return "", err
```

Comment:
Given that we're deleting the mirror dir, should we clone again by default? or is that already handled for us?

Reply:
The checkout hook will automatically retry if it doesn't have a fatal error, I figure we can rely on this behaviour?

Reply:
Sounds good to me